src/averageAges.js

```diff
@@ -54,8 +72,26 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+function calculateAverageAgeDiff(people, onlyWithSon = false) {
+  const women = people.filter((person) => person.sex === 'f')
+    .map((w) => w.name);
+  const kidHaveMother = people.filter((p) => women.includes(p.mother));
+  const onlySons = kidHaveMother.filter((p) => p.sex === 'm');
+
+  const typeArrForAverage = onlyWithSon
+    ? onlySons
+    : kidHaveMother;
+
+  const differense = typeArrForAverage.map((kid) => {
+    const mother = people.find((w) => w.name === kid.mother);
+
+    return kid.born - mother.born;
+  });
+
+  const averageDiffernce = differense.reduce((a, b) => (a + b))
```

Comment:
```suggestion
  const averageDiffernce = differense.reduce((sum, age) => (sum + age))
```
Pay more attention to variable names